util/util.go

```diff
@@ -69,6 +69,17 @@ func RandomString(n int) string {
 	return string(result)
 }
 
+// GetMachinesForCluster returns a list of machines associated with the cluster.
+func GetMachinesForCluster(ctx context.Context, c client.Client, cluster *clusterv1.Cluster) (*clusterv1.MachineList, error) {
+	var machines clusterv1.MachineList
+	if err := c.List(ctx, &machines, client.MatchingLabels{
```

Comment:
Also need to match on cluster namespace

Reply:
Oops, I deleted that filter by accident. Thanks, will fix